### PR TITLE
school subscriptions page

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Create a School Subscriptions page for school admins.

## WHY
See Notion card: https://www.notion.so/quill/Add-a-School-Subscriptions-tab-to-the-administrator-dashboard-502ed1e916b34be28fe1d49ecfbc1886#beff125313a74d0194cb7f8b2381010e.

## HOW
Create a new controller endpoint with the necessary data and a new container component that only renders the relevant parts of the page, passing in the school-specific data. Note: this is going to need to include work from [Brendan's branch](https://github.com/empirical-org/Empirical-Core/commit/f7ec58847b0eea63e67b8fd536915f4bf3eb8a1b) for the purchase buttons to work.

### Screenshots
Screen Shot 2022-05-18 at 12 47 59 PM
Screen Shot 2022-05-18 at 12 48 07 PM

### Notion Card Links
https://www.notion.so/quill/Add-a-School-Subscriptions-tab-to-the-administrator-dashboard-502ed1e916b34be28fe1d49ecfbc1886

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

**Review by eadams17 — APPROVED** [review]

**Review by cissyxyu — APPROVED** [review]

File: services/QuillLMS/app/controllers/subscriptions_controller.rb [hunk 1]
```diff
@@ -83,6 +103,11 @@ def retrieve_stripe_subscription
     end
   end
 
+  private def set_stripe_variables
```
Comment (brendanshean) [inline services/QuillLMS/app/controllers/subscriptions_controller.rb:106]: I like this extraction

File: services/QuillLMS/spec/controllers/subscriptions_controller_spec.rb [hunk 2]
```diff
@@ -101,6 +101,34 @@
...
+    let!(:schools_admins1) { create(:schools_admins, school: school1) }
+    let!(:schools_admins2) { create(:schools_admins, user: schools_admins1.user, school: school2) }
+    let!(:subscription1) { create(:subscription, account_type: Subscription::SCHOOL_PAID ) }
+    let!(:school_subscription1) { create(:school_subscription, subscription: subscription1, school: school1) }
+    let!(:subscription2) { create(:subscription, account_type: Subscription::SCHOOL_PAID, expiration: '2020-01-1'.to_date ) }
+    let!(:school_subscription2) { create(:school_subscription, subscription: subscription2, school: school2) }
+
+    before { allow(controller).to receive(:current_user) { schools_admins1.user } }
+
+    describe '#school_admin_subscriptions' do
```
Comment (brendanshean) [inline services/QuillLMS/spec/controllers/subscriptions_controller_spec.rb:116]: circleci is reporting a failing spec here.

File: services/QuillLMS/spec/controllers/subscriptions_controller_spec.rb [hunk 3]
```diff
@@ -101,6 +101,34 @@
...
+      it 'should set the instance variables' do
+        get :school_admin_subscriptions, params: { :format => 'json' }
+        expect(JSON.parse(response.body)['user_associated_school_id']).to eq schools_admins1.user.school
```
Comment (brendanshean) [inline services/QuillLMS/spec/controllers/subscriptions_controller_spec.rb:119]: After confirming spec failure locally, I noticed that `schools_admins1.user.school` is nil.  Is this expected?

**Review by brendanshean — APPROVED** [review]
Just a couple of comments on a spec failure but nothing that should require re-review.  Nice work on this!

**Pull request merged**